### ISSUE TITLE
Maven multi module project (error: generating report: processor's is already initialized!)

### ISSUE THREAD

**Issue by remcocats** [bsorrentino/maven-confluence-plugin#248]
When I run the confluence maven plugin 7.0-rc2 deploy goal I get the error:  
error: generating report: processor's is already initialized!
On the second module of the multi module project. The first one is putting the informatie in confluence.
When I configure the plugin only on the modules or only in the pom where the modules are defined I get the same error.
Does any one else has this problem or do I have to set some configuration.

**Comment by bsorrentino** [comment]
Hi @remcocats 

thanks for feedback I'll check it asap

However I use the plugin in the multimodule project, using the very useful maven parameter `-pl, --projects`.

options | description
---|---
&#x2011;pl,&nbsp;&#x2011;&#x2011;projects&nbsp;\<arg> |  Comma-delimited list of specified reactor projects to build instead of all projects. A project can be specified by [groupId]:artifactId or by its relative path. |

Simply add plugin in project(s) of interest and run:

```
mvn -pl <module name1>[,<module name N>] confluence-reporting:deploy 
```

**Comment by sebb840** [comment]
I have the same error when I use the xml tags <executions>. With the version 6.20, I've not this error. 
When I see the code source of MarkDownProcessor.init() method, what do you think about this patch ?

  	public MarkdownProcessor init( ) {
  	  	if( processor==null ) 
  	  	{
  	  	  	processor =  MarkdownProcessor.load( getName() );                
  	  	}
  	  	return processor;
  	}

you can reproduce error with the configuration : 
  	
  	<plugin>
  		<groupId>org.bsc.maven</groupId>
		<artifactId>confluence-reporting-maven-plugin</artifactId>
  		<version>7.0</version>    					
  		<executions>
  			<execution>
  				<id>exec1</id>
  				<goals>
					<goal>deploy</goal>
				</goals>
  				<phase>validate</phase>  						
  				<configuration>
	  	  			<title>Publish first</title>
	 		 		<labels>		               			 
  	  		  	  	  	<label>label1</label>
	  	          			<label>label2</label>                		
		     			</labels>
				</configuration>
  			</execution>
  			<execution>
  				<id>exec2</id>
  				<goals>
					<goal>deploy</goal>
				</goals>
  				<phase>validate</phase>  						
  				<configuration>
					<title>Publish second</title>
	   		 		<labels>
               					<label>label1</label>
  	  	               			<label>label3</label>                		
		     			</labels>
				</configuration>
			</execution>
  		</executions>  							  			
	</plugin>		

The stacktrace 
	
	org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.bsc.maven:confluence-reporting-maven-plugin:7.0:deploy (exec2) on project myProject: error generating report
	    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:213)
	    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
	    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
	    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
	    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
	    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:51)
	    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
	    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:309)
	    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
	    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
	    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
	    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
	    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
	    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
	    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:64)
	    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
	    at java.lang.reflect.Method.invoke (Method.java:564)
	    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
	    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
	    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
	    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
	Caused by: org.apache.maven.plugin.MojoExecutionException: error generating report
	    at org.bsc.mojo.AbstractBaseConfluenceMojo.execute (AbstractBaseConfluenceMojo.java:384)
	    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:134)
	    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
	    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
	    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
	    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
	    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
	    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:51)
	    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
	    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:309)
	    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
	    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
	    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
	    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
	    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
	    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
	    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:64)
	    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
	    at java.lang.reflect.Method.invoke (Method.java:564)
	    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
	    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
	    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
	    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
	Caused by: java.lang.IllegalStateException: processor's is already initialized!
	    at org.bsc.markdown.MarkdownProcessor$Shared.init (MarkdownProcessor.java:86)
	    at org.bsc.mojo.ConfluenceDeployMojo.execute (ConfluenceDeployMojo.java:276)
	    at org.bsc.mojo.AbstractBaseConfluenceMojo.execute (AbstractBaseConfluenceMojo.java:377)
	    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:134)
	    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
	    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
	    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
	    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
	    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
	    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:51)
	    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
	    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:309)
	    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
	    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
	    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
	    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
	    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
	    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
	    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:64)
	    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
	    at java.lang.reflect.Method.invoke (Method.java:564)
	    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
	    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
	    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
	    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
	[ERROR]
	[ERROR]
	[ERROR] For more information about the errors and possible solutions, please read the following articles:
	[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

**Comment by bsorrentino** [comment]
Hi @sebb840 

I think that your patch (better call it solution) is perfect. I wanted check multiple initialisation but it was useful in development phase now is definetevely useless

**Comment by bsorrentino** [comment]
fix released for test in version `7.1-SNAPSHOT`

I'll arrange a new release asap

**Comment by bsorrentino** [comment]
released `7.1` with fix
